raiden/network/proxies/secret_registry.py

```diff
@@ -79,6 +81,10 @@ def register_secret(self, secret: typing.Secret):
             contract=pex(self.address),
         )
 
+        with self.lock:
+            return self._register_secret(secret)
```

Comment:
This is not sufficient, the secret will not be registered in parallel but all calls to `register_secret` will send a transaction. The goal here is to save on the number of transactions.

>  The task which obtained the lock sends the transaction, the others wait on the result

Reply:
ah, ok, totally makes sense. I'll check how it's done in `ChannelManager` proxy and make something similar.